test/e2e/tests/test_elasticsearch_domain.py

```diff
@@ -49,14 +52,40 @@ def get_resource_arn(self, resource: Dict):
     return resource['status']['ackResourceMetadata']['arn']
 
 
+def wait_for_create_or_die(es_client, resource, timeout):
+    aws_res = es_client.describe_elasticsearch_domain(DomainName=resource.name)
+    while aws_res['DomainStatus']['Processing'] == True:
+        if datetime.datetime.now() >= timeout:
+            pytest.fail("Timed out waiting for ES Domain to get DomainStatus.Processing == False")
+        time.sleep(CREATE_WAIT_INTERVAL_SLEEP_SECONDS)
+
+        aws_res = es_client.describe_elasticsearch_domain(DomainName=resource.name)
+
+    return aws_res
+
+
+def wait_for_delete_or_die(es_client, resource, timeout):
+    while True:
+        if datetime.datetime.now() >= timeout:
+            pytest.fail("Timed out waiting for ES Domain to being deleted in AES API")
+        time.sleep(DELETE_WAIT_INTERVAL_SLEEP_SECONDS)
+
+        try:
+            aws_res = es_client.describe_elasticsearch_domain(DomainName=resource.name)
+            if aws_res['DomainStatus']['Deleted'] == False:
+                pytest.fail("DomainStatus.Deleted is False for ES Domain that was deleted.")
+        except es_client.exceptions.ResourceNotFoundException:
+            break
+
+
 @service_marker
 @pytest.mark.canary
 class TestDomain:
     def test_create_delete_7_9(self, es_client):
-        resource_name = "my-es-domain"
+        resource = Domain("my-es-domain",1,0,False,False,[])
```

Comment:
I actually find this harder-to-read than either having separate variables (like `data_node_count`, `master_node_count`, etc), using a `dict` with named keys or having an actual `object` with named fields.

Reply:
TIL!